src/main/java/io/cryostat/recordings/RecordingTargetHelper.java

```diff
@@ -156,6 +172,115 @@ public IRecordingDescriptor startRecording(
+    }
+
+    public Future<Void> deleteRecording(
+            ConnectionDescriptor connectionDescriptor, String recordingName) {
+        CompletableFuture<Void> future = new CompletableFuture<>();
+        try {
+            String targetId = connectionDescriptor.getTargetId();
+            targetConnectionManager.executeConnectedTask(
+                    connectionDescriptor,
+                    connection -> {
+                        Optional<IRecordingDescriptor> descriptor =
+                                connection.getService().getAvailableRecordings().stream()
+                                        .filter(
+                                                recording ->
+                                                        recording.getName().equals(recordingName))
+                                        .findFirst();
+                        if (descriptor.isPresent()) {
+                            connection.getService().close(descriptor.get());
+                            reportService.delete(connectionDescriptor, recordingName);
+                        } else {
+                            throw new RecordingNotFoundException(targetId, recordingName);
+                        }
+                        return null;
+                    });
+            future.complete(null);
+        } catch (Exception e) {
+            future.completeExceptionally(e);
+        }
+        return future;
+    }
+
+    public Future<SnapshotMinimalDescriptor> createSnapshot(JFRConnection connection) {
+        CompletableFuture<SnapshotMinimalDescriptor> future = new CompletableFuture<>();
+        try {
+            IRecordingDescriptor descriptor = connection.getService().getSnapshotRecording();
+
+            String rename =
+                    String.format("%s-%d", descriptor.getName().toLowerCase(), descriptor.getId());
+
+            RecordingOptionsBuilder recordingOptionsBuilder =
+                    recordingOptionsBuilderFactory.create(connection.getService());
+            recordingOptionsBuilder.name(rename);
+
+            connection
+                    .getService()
+                    .updateRecordingOptions(descriptor, recordingOptionsBuilder.build());
+
+            future.complete(new SnapshotMinimalDescriptor(rename, descriptor));
+        } catch (Exception e) {
+            future.completeExceptionally(e);
+        }
+        return future;
+    }
+
+    public Future<Boolean> verifySnapshot(
+            ConnectionDescriptor connectionDescriptor, String snapshotName) {
+        CompletableFuture<Boolean> future = new CompletableFuture<>();
+        try {
+            Optional<InputStream> snapshotOptional =
+                    this.getRecording(connectionDescriptor, snapshotName).get();
+            if (snapshotOptional.isEmpty()) {
+                throw new SnapshotCreationException(snapshotName);
+            } else if (!snapshotIsReadable(connectionDescriptor, snapshotOptional.get())) {
```

Comment:
The result of `snapshotOptional.get()` is an `InputStream` that we acquired within this method, so we should also be careful to close this stream within this method. It looks a little hairy, but maybe that looks something like:


```java
Optional<InputStream> opt = this.getRecording().get();
if (opt.isEmpty() {
  throw
} else {
  try (InputStream stream = opt.get()) {
    if (!snapshotIsReadable(connDesc, stream)) {
      this.deleteRecording(connDesc, snapshotName).get();
      future.complete(false);
    } else {
      future.complete(true);
    }
  }
}
```